src/lib/reasoners/theory.ml

```diff
@@ -474,20 +414,20 @@ module Main_Default : S = struct
         assert false
 
       | Pinfinity | Minfinity | Limit _ ->
-        (* We stop optimizing the split [opt_split] in this case, but
+        (* We stop optimizing the objective function [obj] in this case, but
            we continue to produce a model if the flag [for_model] is up. *)
         if for_model then
-          aux env sem_facts acc_choices []
+          aux env sem_facts acc_choices ([], 0) []
```

Comment:
I believe this is a behavior change — previously we would keep optimizing further objectives while doing model generation, but now we no longer does (this call to `aux` goes immediately to `generate_choices`). Is that intentional?

Reply:
Also as far as I can tell, this means that the case splits in the middle of intervals and the such that we create for infinite/limit cases are never used.